jina/orchestrate/deployments/__init__.py

```diff
@@ -107,14 +85,14 @@ def head_host(self) -> str:
         """Get the host of the HeadPod of this deployment
         .. # noqa: DAR201
         """
-        return self.head_args.host
+        return self.head_args.host if self.head_args else None
 
     @property
     def head_port(self):
```

Comment:
we would need then another property for `ports()`? Should we have inheritance in Deployments?

Reply:
you mean we should expose sth like `worker_ports`? Yeah that makes sense. I dont think we need inheritance though? Every deployment should have worker_ports. Just their usage in the Gateway and other places differ, but thats not the issue of the `Deployment` class?

Reply:
added a new `ports` property which can be used from the Flow and other places (like tests etc)